packages/patterns/src/patterns/patternMatchers.js

```diff
@@ -1651,13 +1651,26 @@ const makePatternKit = () => {
       limits ? makeLimitsMatcher('match:symbol', [limits]) : SymbolShape,
     record: (limits = undefined) =>
       limits ? M.recordOf(M.any(), M.any(), limits) : RecordShape,
+    // The pattern for a CopyRecord with a fixed quantity of heterogeneous
+    // entry types (a Struct) is merely a hardened object with the respective
+    // key name to pattern entries.
+    // For example, the TypeScript { x: string, y: number } has the pattern
+    // harden({ x: M.string(), y: M.number() }).
     array: (limits = undefined) =>
       limits ? M.arrayOf(M.any(), limits) : ArrayShape,
+    // The pattern for a CopyArray with a fixed quantity of heterogeneous value
+    // types (a Tuple) is merely a hardened array of pattern types.
+    // For example, the TypeScript [string, number] has the pattern
+    // harden([ M.string(), M.number() ]).
     set: (limits = undefined) => (limits ? M.setOf(M.any(), limits) : SetShape),
     bag: (limits = undefined) =>
       limits ? M.bagOf(M.any(), M.any(), limits) : BagShape,
     map: (limits = undefined) =>
       limits ? M.mapOf(M.any(), M.any(), limits) : MapShape,
+    // The pattern for a CopyMap with a fixed quantity of heterogeneous entry
+    // types is merely a CopyMap of corresponding key and value patterns, for
+    // example,
+    // makeCopyMap([[M.string(), M.number()], [M.number(), M.string()]]).
```

Comment:
No. The keys must be keys. Only the values can be non-key patterns.

Reply:
Rewrote record and map to clarify this distinction.